libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Network/Network/Queue/CSFNetwork.m

```diff
@@ -417,6 +417,7 @@ - (void)delegate_networkCanceledAction:(CSFAction*)action {
             [delegate network:self didCancelAction:action];
         }
     }];
+    [action dequeueNetwork:self];
```

Comment:
> You think it makes more sense for CSFAction to own dequeueing?

I do.  First, the delegate methods are triggered by actions that occur in `CSFAction` anyway.  And second, this is really a matter of `CSFAction` cleaning itself up anyway—it holds all of the pertinent logic related to the lifecycle of its request, so it makes sense that it would manage the network it's holding, after it's done.  Third, it will consolidate the code into one place, and that place is IMO a more logical place to look for action lifecycle cleanup.